demos/wallet/send.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/kaspanet/kaspad/ecc"
+	"github.com/kaspanet/kaspad/txscript"
+	"github.com/kaspanet/kaspad/util"
+	"github.com/kaspanet/kaspad/util/daghash"
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/kaspanet/kasparov/kasparovd/apimodels"
+	"github.com/pkg/errors"
+)
+
+const feeSompis uint64 = 1000
+
+func send(conf *sendConfig) error {
+	toAddress, err := util.DecodeAddress(conf.ToAddress, util.Bech32PrefixUnknown)
+	if err != nil {
+		return err
+	}
+
+	privateKey, publicKey, err := parsePrivateKey(conf.PrivateKey)
+	if err != nil {
+		return err
+	}
+
+	fromAddress, err := util.NewAddressPubKeyHashFromPublicKey(publicKey.SerializeCompressed(), toAddress.Prefix())
+	if err != nil {
+		return err
+	}
+
+	utxos, err := getUTXOs(conf.KasparovAddress, fromAddress.String())
+	if err != nil {
+		return err
+	}
+
+	sendAmountSompi := uint64(conf.SendAmount * util.SompiPerKaspa)
+	totalToSend := sendAmountSompi + feeSompis
+
+	selectedUTXOs, change, err := selectUTXOs(utxos, totalToSend)
+	if err != nil {
+		return err
+	}
+
+	msgTx, err := generateTx(privateKey, selectedUTXOs, sendAmountSompi, change, toAddress, fromAddress)
+	if err != nil {
+		return err
+	}
+
+	err = sendTx(conf, msgTx)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Transaction was sent successfully")
+	fmt.Printf("Transaction ID: \t%s", msgTx.TxID())
+
+	return nil
+}
+
+func parsePrivateKey(privateKeyHex string) (*ecc.PrivateKey, *ecc.PublicKey, error) {
+	privateKeyBytes, err := hex.DecodeString(privateKeyHex)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Error parsing private key hex")
+	}
+	privateKey, publicKey := ecc.PrivKeyFromBytes(ecc.S256(), privateKeyBytes)
+	return privateKey, publicKey, nil
+}
+
+func selectUTXOs(utxos []*apimodels.TransactionOutputResponse, totalToSpend uint64) (
+	selectedUTXOs []*apimodels.TransactionOutputResponse, change uint64, err error) {
+
+	selectedUTXOs = []*apimodels.TransactionOutputResponse{}
+	totalValue := uint64(0)
+
+	for _, utxo := range utxos {
+		if utxo.IsSpendable == nil || !*utxo.IsSpendable {
+			continue
+		}
+
+		selectedUTXOs = append(selectedUTXOs, utxo)
+		totalValue += utxo.Value
+
+		if totalValue >= totalToSpend {
+			break
+		}
+	}
+
+	if totalValue < totalToSpend {
+		return nil, 0, errors.Errorf("Insufficient funds for send: %f required, while only %f available",
+			float64(totalToSpend)/util.SompiPerKaspa, float64(totalValue)/util.SompiPerKaspa)
```

Comment:
This is likely to print a weird float-y number like 2.1023445130464. Please format the floats to only have a few digits after the decimal point. I recommend two digits.

Reply:
It's going to be exactly up to 8 decimal points, and this is also the number you want to get.